var/spack/repos/builtin/packages/migraphx/package.py

```diff
@@ -42,10 +43,11 @@ def url_for_version(self, version):
     depends_on('nlohmann-json', type='link')
     depends_on('msgpack-c', type='link')
     depends_on('half@1.12.0', type='link')
-    depends_on('py-pybind11', type='build')
+    depends_on('py-pybind11', type='build', when='@:4.0.0')
```

Comment:
```suggestion
    depends_on('py-pybind11', type='build')
```
This is not strictly necessary, the next line adds an additional constraint just for 4.1.0+.